plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -30,92 +28,41 @@ const getContext = ( ns?: string ) =>
 type Store = typeof productGallery & StorePart< ProductGallery >;
 const { state, actions } = store< Store >( 'woocommerce/product-gallery' );
 
-/**
- * Product Gallery supports two contexts:
- * - on-page gallery - may display subset of images.
- * - dialog gallery - displays all of the images.
- * Function returns images per current context.
- */
-const getCurrentImages = ( context: ProductGalleryContext ) => {
-	const { isDialogOpen } = context;
-	return context[
-		isDialogOpen ? 'dialogVisibleImagesIds' : 'visibleImagesIds'
-	];
+const getImageIndex = (): number => {
+	return state.imageIds.indexOf( state.imageId );
 };
 
-const getImageIndex = ( context: ProductGalleryContext, imageId: string ) => {
-	const imagesIds = getCurrentImages( context );
-	return imagesIds.indexOf( imageId );
-};
-
-const disableArrows = (
-	context: ProductGalleryContext,
-	newImageNumber: number
-) => {
-	const imagesIds = getCurrentImages( context );
+const getArrowsState = ( imageNumber: number, totalImages: number ) => ( {
```

Comment:
Same here. Feels like this should belong on the state object with the other getters? What do you think? These are derived values from state. We could then remove the `totalImages` param

Reply:
We can't do that. Getter cannot accept any arguments and we still need to provide "new image number" to calculate the state.